src/main/java/edu/cornell/kfs/module/purap/document/authorization/IWantDocumentPresentationController.java

```diff
@@ -148,13 +149,15 @@ public Set<String> getEditModes(Document document) {
         }
         
         editModes.add(CUPurapConstants.I_WANT_DOC_EDIT_PROC_NET_ID);
-        editModes.add(CUPurapConstants.I_WANT_DOC_EDIT_PROCUREMENT_ASSISTANT_NET_ID);
         editModes.add(CUPurapConstants.IWNT_DOC_DISPLAY_NOTE_OPTIONS);
         
         if (isContractFunctionalityEnabled()) {
             editModes.add(CUPurapConstants.IWNT_DOC_DISPLAY_CONTRACT_TAB);
             editModes.add(CUPurapConstants.IWNT_DOC_EDIT_CONTRACT_INDICATOR);
-
+            if (workflowDocument.isEnroute() && workflowDocument.getCurrentNodeNames().contains(
+                    IWantRouteNodes.PROCUREMENT_CONTRACT_ASSISTANT)) {
+                editModes.add(CUPurapConstants.I_WANT_DOC_EDIT_PROCUREMENT_ASSISTANT_NET_ID);
+            }
```

Comment:
Could this code go in the canEditProcurementAssistantNetId method?

Reply:
I realize that the related authorizer class already contains many of the route-node-based and workflow-status-based checks for the other edit modes. However, I was trying to set this part up to put most of the non-person-specific checks into the presentation controller, which I think is how KFS generally expects edit modes and action authorization to be handled.

I could alternatively put these checks into the authorizer class for consistency with our other IWNT setup. If you have a strong opinion about which approach would be better, please let me know.

Reply:
I am fine doing it this way if it aligns with our best practices.